### DISCUSSION TITLE
Creating database host for nodes

### DISCUSSION THREAD

**Discussion by opiepl** [pterodactyl/panel#2856] [Support] [answered]
I am trying to make database host for nodes (externally). In docs I found that "If your database is on a different host than the one where your Panel or Daemon is installed make sure to use the IP address of the machine the Panel is running on.". So I bought new vps, than installed mysql, than I pasted this commands:
USE mysql;
CREATE USER 'pterodb'@'{PANEL ADRESS}' IDENTIFIED BY '{PASSWORD}';
GRANT ALL PRIVILEGES ON *.* TO 'pterodb'@'{PANEL ADRESS}' WITH GRANT OPTION;
FLUSH PRIVILEGES;

Than I edited my.cnf to allow connecting externally, and finally when I tried add the database host in panel, I saw this:
There was an error while trying to connect to the host or while executing a query: "SQLSTATE[HY000] [2002] Connection refused (SQL: SELECT 1 FROM dual)"

**Comment by schrej** [comment]
The `connection refused` error means that the database server didn't respond, there is nothing listening on that port.
You either made a mistake when editing the `my.cnf`, or didn't restart the server after doing so.
It could also be a firewall that blocks connections on that port.

**Comment by opiepl** [comment, marked as answer]
Restarting the mysql server helped. I would made a tutorial how to make database host for nodes. Command to restart: service mysql restart
